### PR TITLE
Fixed #26153 -- Reallowed Q-objects from get_extra_descriptor_filter()

### PR DESCRIPTION
**Pull request by akaariai**

### CODE REVIEW

File: django/db/models/fields/related.py [hunk 1]
```diff
@@ -340,8 +341,13 @@ def get_reverse_related_filter(self, obj):
...
             for lh_field, rh_field in self.related_fields
         }
-        base_filter.update(self.get_extra_descriptor_filter(obj) or {})
-        return base_filter
+        descriptor_filter = self.get_extra_descriptor_filter(obj)
+        base_q = Q(**base_filter)
+        if isinstance(descriptor_filter, dict):
+            return base_q & Q(**descriptor_filter)
+        elif descriptor_filter:
+            return base_q & descriptor_filter
```
Comment (timgraham) [inline django/db/models/fields/related.py]: Is this branch tested? No tests seem to fail (I tired SQLite & PG) if it's removed.

**Comment by timgraham** [comment]
Could you add a mention in the 1.9.2 release notes?

File: django/db/models/fields/related.py (hunk 1 above)
Comment (akaariai) [inline django/db/models/fields/related.py]: Good catch! The select_related() in the added test prevented the test from testing this branch.

**Comment by akaariai** [comment]
I can't repeat the test failure locally.

**Comment by timgraham** [comment]
buildbot, test this please.

**Comment by timgraham** [comment]
merged in 353aecbf8c1a8cc6f3985149e2895d49e53dfc1c, thanks!

**Pull request closed without merging**